### ISSUE TITLE
[java] CognitiveComplexity returns None when Cyclomatic returns float value

### ISSUE THREAD

**Issue by lyriccoder** [pmd/pmd#3327]
**Affects PMD Version:** 
6.35.0

**Description:**
Cognitive Rule for java fails for many cases for Qualitas dataset. There are 1685946 methods which were returned with Cyclomatic Rule. 1068191 values with Cognitive Complexity were `NaN` values.
So, only 617755 values were not None

**Steps to reproduce:**

1.  Download QualitasCorpus-20130901r dataset  
2.  Extract it as it is described inside archive
3. Download pmd 6.35.0
4. Run pmd with the rules (see an attached file)
5. Compare results

**Run through cmd** 

```
./pmd-bin-6.35.0/bin/run.sh pmd -cache ./_tmp/cache -d /dataset/01 -R ruleset.xml -f csv > tmp_out.csv
```

```
No LSB modules are available.
Distributor ID: Ubuntu
Description: Ubuntu 18.04.5 LTS
Release: 18.04
Codename: bionic
```

```
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
CPU(s):              8
On-line CPU(s) list: 0-7
Thread(s) per core:  2
Core(s) per socket:  4
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               85
Model name:          Intel(R) Xeon(R) Gold 6151 CPU @ 3.00GHz
Stepping:            4
CPU MHz:             3000.000
BogoMIPS:            6000.00
Hypervisor vendor:   KVM
Virtualization type: full
L1d cache:           32K
L1i cache:           32K
L2 cache:            1024K
L3 cache:            25344K
NUMA node0 CPU(s):   0-7
```

I have noticed that Cognitive Rule fails for 1 line methods frequently.
Could you please check all the cases? I have attached a zip file with Nan values for Cognitive Rule. You can find files in qualitas dataset url

**Comment by lyriccoder** [comment]
Here is dataset 20130901r:
http://qualitascorpus.com/download/
Here is the Rules.xml:
[ruleset.txt](https://github.com/pmd/pmd/files/6609477/ruleset.txt)
Here is the dataset with failed rows:
[wrong_pmd.tar.gz](https://github.com/pmd/pmd/files/6609511/wrong_pmd.tar.gz)

**Comment by adangel** [comment]
Hi @lyriccoder , thanks for the report.
I don't think, that's a bug - it works as expected.

Consider the following sample class:

```java
public class Container {
    private String data;
    public void setData(String data) {
        this.data = data;
    }
}
```

If you analyze that with PMD and your ruleset, the result is as follows:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 http://pmd.sourceforge.net/report_2_0_0.xsd" version="6.35.0" timestamp="2021-06-17T08:51:21.842">
    <file name="/home/andreas/PMD/source/pmd-it/issue-3327-cognitive-complexity/samplesrc/Container.java">
        <violation beginline="3" endline="5" begincolumn="12" endcolumn="5" rule="NcssCount" ruleset="Design" class="Container" method="setData" externalInfoUrl="https://pmd.github.io/pmd-6.35.0/pmd_rules_java_design.html#ncsscount" priority="3">
The method 'setData(String)' has a NCSS line count of 2.
        </violation>
        <violation beginline="3" endline="5" begincolumn="12" endcolumn="5" rule="CyclomaticComplexity" ruleset="Design" class="Container" method="setData" externalInfoUrl="https://pmd.github.io/pmd-6.35.0/pmd_rules_java_design.html#cyclomaticcomplexity" priority="3">
The method 'setData(String)' has a cyclomatic complexity of 1.
        </violation>
        <violation beginline="3" endline="5" begincolumn="12" endcolumn="5" rule="NPathComplexity" ruleset="Design" class="Container" method="setData" externalInfoUrl="https://pmd.github.io/pmd-6.35.0/pmd_rules_java_design.html#npathcomplexity" priority="3">
The method 'setData(String)' has an NPath complexity of 1, current threshold is 1
        </violation>
    </file>
</pmd>
```

We don't report in this case a cognitive complexity for this method, because the cognitive complexity is 0 for this trivial setter.

You can't currently change the report-level for cognitive complexity to 0 ("Constraint violated on property value '0' (Should be positive)"), but if you could, you'd get the following result:

    The method 'setData(String)' has a cognitive complexity of 0, current threshold is 0

So in summary: The rule cognitive complexity works as expected - it does not report methods, that are not complex.
